### PR TITLE
added vmm3 generator for ess readout supporting fuzzing

### PR DESCRIPTION
**Pull request by mortenjc**
### Issue reference / description

The branch you merge from should already reference an event-formation-unit github ticket number. You can add a descriptive title, but if an issue is referenced, you don't have to.

## Checklist for submitter

- [ ] Check for conflict with integration test
- [ ] Unit tests pass

---

## Nominate for Group Code Review (Anyone can nominate it)
Indicate if you think the code should be reviewed in a Thursday code review session.

- [ ] Recommend for group code review

Also, nominate it on the code_review Slack channel.

### CODE REVIEW

**Comment by mortenjc** [comment]
PR build fails with: "fatal error: error in backend: IO failure on output stream: No space left on device"

File: src/modules/readout/vmm3/generators/udpgen_vmm3.cpp [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      usleep(Settings.SpeedThrottle);
+    }
+    Packets++;
+    TotalPackets++;
+    if (Settings.PktThrottle) {
+      if (Packets % Settings.PktThrottle == 0) {
+        usleep(10);
+      }
+    }
+    if (Settings.NumberOfPackets != 0 and Packets >= Settings.NumberOfPackets) {
```
Comment (amues) [inline src/modules/readout/vmm3/generators/udpgen_vmm3.cpp:87]: Maybe I am missing something, but are both `Packets` and `TotalPackets` required? Why is `Packets` set to zero right before breaking the loop, as it is a local variable and not used afterwards?

**Review by amues — APPROVED** [review]

File: src/modules/adc_readout/test/UDPClientTest.cpp [hunk 2]
```diff
@@ -19,24 +19,27 @@ class UDPClientTest : public ::testing::Test {
...
-        ++PacketsHandled;
-      };
-  UDPClient TestClient(Service, "0.0.0.0", ListenOnPort, PacketHandler);
-  Server.startPacketTransmission(1, 0);
-  Service->run_for(100ms);
-  EXPECT_EQ(BytesReceived, BytesToTransmit);
-  EXPECT_EQ(PacketsHandled, 1);
-}
+
+/// MJC disabled this test as it periodically fails and creates
```
[the remark this answers is not in the document]
Comment (mortenjc) [reply src/modules/adc_readout/test/UDPClientTest.cpp:23]: Not other than ignorance. I'll fix this

File: src/modules/readout/vmm3/generators/udpgen_vmm3.cpp (hunk 1 above)
Comment (mortenjc) [reply src/modules/readout/vmm3/generators/udpgen_vmm3.cpp:87]: I'll have a look anf fix this. This is copy and paste from udpgen_loki

**Pull request merged**